### DISCUSSION TITLE
Increase sql data source query textbox height

### DISCUSSION THREAD

**Discussion by tve** [grafana/grafana#37004] [Data Sources]
### What Grafana version are you using?
6.0 beta 2
### What datasource are you using?
postgres

Is there a chance to get a small button to increase the vertical height of the query textbox? For more sophisticated queries it's pretty painful to edit with just ~10 lines visible. E.g., to produce the contents of a table using the postgres crosstab function to get a pivot:
![image](https://user-images.githubusercontent.com/39480/52980006-02afed00-338d-11e9-99e1-8d2231751d4e.png). A simple "+" icon one can click in the upper-right would do the trick, I believe. Or allow the textbox to be resized in the browser.
Thanks!

**Comment by marefr** [comment]
Given the new panel editor layout in Grafana 6 I think we can increase the maximum lines shown.

Seems to be quite easy to change the maximum number of lines from 10 to something else (20?) per each sql datasource, for example with postgres:

```diff
diff --git a/public/app/plugins/datasource/postgres/partials/query.editor.html b/public/app/plugins/datasource/postgres/partials/query.editor.html
index b5e2b5d..9d454ee 100644
--- a/public/app/plugins/datasource/postgres/partials/query.editor.html
+++ b/public/app/plugins/datasource/postgres/partials/query.editor.html
@@ -3,7 +3,7 @@
   <div ng-if="ctrl.target.rawQuery">
     <div class="gf-form-inline">
       <div class="gf-form gf-form--grow">
-        <code-editor content="ctrl.target.rawSql" datasource="ctrl.datasource" on-change="ctrl.panelCtrl.refresh()" data-mode="sql">
+        <code-editor content="ctrl.target.rawSql" datasource="ctrl.datasource" on-change="ctrl.panelCtrl.refresh()" data-mode="sql" max-lines="20">
         </code-editor>
       </div>
     </div>
```

**Comment by gabor** [comment]
in the current Grafana version, the query editor has a button at the bottom-right corner that will enlarge the editor to the largest possible size. this should solve the problem.

if it does not solve it, please open a new github issue with the problem and also describe how the current solution is insufficient. thanks!
